Add support for metric prefix

skygeo opened the pull request:
Prepend every metrics with a custom prefix set in the toml configuration file.

skygeo commented:
Hi @kylebrandt @mjibson @gbrayut @Dieterbe
Is there a process for reviews? Or raising features like the ones proposed here?

Also, I am not sure why the tests are failing here, it seems unrelated. 

Let me know if I can help in any ways.
Many Thx

captncraig commented on cmd/scollector/conf/conf.go:
```diff
@@ -63,6 +63,9 @@ type Conf struct {
 	UseSWbemServicesClient bool
 
+	// Prefix prepended to all metrics path
+	Prefix string
```
Let's call this "MetricPrefix" to be more explicit.

captncraig commented on cmd/scollector/collectors/collectors.go:
```diff
@@ -296,6 +301,9 @@ func AddTS(md *opentsdb.MultiDataPoint, name string, ts int64, value interface{}
 // automatically added. If the value of the host key is the empty string, it
 // will be removed (use this to prevent the normal auto-adding of the host tag).
 func Add(md *opentsdb.MultiDataPoint, name string, value interface{}, t opentsdb.TagSet, rate metadata.RateType, unit metadata.Unit, desc string) {
+	if Prefix != "" {
```
This code needs to be in `AddTS` instead of `Add`, so it works for everything.

captncraig commented:
It looks like the build is failing because `gofmt` needs to be run.

skygeo commented:
Hi @captncraig ! Thank's for your input.
+ The argument is now "MetricPrefix" instead of "Prefix".
+ Moved the code in the "AddTS" method instead of "Add" method.
+ Added some documentation.
+ Travis Build now passing.

Let me know if you need anything else.

skygeo commented:
Any news on that? @captncraig  ??

captncraig commented:
Sorry, this looks good now. Only thing left is a mention in  the documentation (https://github.com/bosun-monitor/bosun/blob/master/cmd/scollector/doc.go)

captncraig commented:
Looks great. Thanks.
